cmd/tmpfs/tmpfs.go

```diff
@@ -308,13 +309,20 @@ func main() {
 		flag.Usage()
 	}
 	pkg := a[0]
-	pkfFile, err := os.Open(pkg)
+	pkgFile, err := os.Open(pkg)
 	if err != nil {
 		log.Fatal(err)
 	}
-	defer pkfFile.Close()
+	defer pkgFile.Close()
+
+	var arch *tmpfs.Archive
```

Comment:
This is pretty inflexible. What if for some types we'd rather look for magic numbers instead of extension?

consider this instead:

type decoder func(n string) (arch, error)
var decoders = []decoder{tmpfs.ReadImageGZ, ReadImageCPIO, }
for _, d := range decoders {
a, err = d(pkg)
if err == nil { break}
}

so why do this? Because you can let the decoder decide. The decoder can go by file extension, as in tar files; or look in the cpio for the magic, or whatever it wants. Over time, as we get more decoders this is easier. 

The other problem: what if I want to have a gzip'ed cpio? Your setup here can't do that.

so consider breaking decoding and unpacking apart, in a future release.